### PR TITLE
New tier groups and Tests

### PR DESCRIPTION
**Pull request by ruks**
added new tier groups and testcase
fixed https://wso2.org/jira/browse/APIMANAGER-4374

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+        deleteAPI(APITierTEST_ID, apiPublisherClientUser);
+        deleteAPI(APPTierTEST_ID, apiPublisherClientUser);
+        deleteAPI(RESTierTEST_ID, apiPublisherClientUser);
+        deleteAPI(Before_TierTEST_ID, apiPublisherClientUser);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+
+        deleteAPI(After_TierTEST_ID, apiPublisherClientUser);
```
Comment (chamilaadhi) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: No need to delete the api if  super.cleanUp(); is called afterwards . super.cleanUp() will delete all the apis

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+
+        super.cleanUp();
```
Comment (chamilaadhi) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: No need to delete the api if super.cleanUp(); is called afterwards . super.cleanUp() will delete all the apis

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+    private String newTiersXML;
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+    private APIPublisherRestClient apiPublisherClientUser;
+    private APIStoreRestClient apiStoreClientUser;
+
+    private APIIdentifier APITierTEST_ID;
+    private APIIdentifier APPTierTEST_ID;
+    private APIIdentifier RESTierTEST_ID;
+    private APIIdentifier Before_TierTEST_ID;
+    private APIIdentifier After_TierTEST_ID;
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: variable naming is not consistant

Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: And should beging with lower case later if not a constant

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+                    bufferedReader.close();
+                } catch (IOException e) {
+                    log.warn("Error when closing the buffer reade which used to reed the file:" + fileLocation +
+                            ". Error:" + e.getMessage());
+                }
+            }
+        }
+    }
+
+    protected HttpResponse subscribeToAPI(APIIdentifier apiIdentifier, String applicationName,
```
Comment (chamindias) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: Please add method comments.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java (hunk 2 above)
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: revert changes in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java (hunk 3 above)
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: revert changes already in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java (hunk 4 above)
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: revert changes already in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][] { new Object[] { TestUserMode.SUPER_TENANT_ADMIN },
+                new Object[] { TestUserMode.TENANT_ADMIN }, };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        System.out.println(userMode);
```
Comment (nuwand) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: Don't use sysouts.

Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: already revert changes in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
+        }
+    }
+
+    protected HttpResponse subscribeToAPI(APIIdentifier apiIdentifier, String applicationName,
+            APIStoreRestClient storeRestClient) throws APIManagerIntegrationTestException {
+        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(apiIdentifier.getApiName(),
+                apiIdentifier.getProviderName());
+        subscriptionRequest.setVersion(apiIdentifier.getVersion());
+        subscriptionRequest.setApplicationName(applicationName);
+        if ((apiIdentifier.getTier() != null) && (!apiIdentifier.getTier().equals(""))) {
```
Comment (pamod) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: Should we flip the statement  ' apiIdentifier.getTier().equals("") ? '

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java [hunk 7]
```diff
@@ -0,0 +1,610 @@
...
+            JSONObject response = new JSONObject(responseString);
+
+            applicationKeyBean.setAccessToken(response.getJSONObject("data").getJSONObject("key").
+                    get("accessToken").toString());
+            applicationKeyBean.setConsumerKey(response.getJSONObject("data").getJSONObject("key").
+                    get("consumerKey").toString());
+            applicationKeyBean.setConsumerSecret(response.getJSONObject("data").getJSONObject("key").
+                    get("consumerSecret").toString());
+            return applicationKeyBean;
+        } catch (Exception e) {
```
Comment (pamod) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: Any reason why we capture a generic exception here ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java (hunk 6 above)
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: already revert changes in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java (hunk 7 above)
Comment (ruks) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/throttling/NewThrottleTiersAndLoadOldTiersTestCase.java]: already revert changes in commit: https://github.com/wso2/product-apim/commit/a75c4880b08d984d9a0e971387c0d6a07ef2c886

**Pull request merged**